template-vue-ts/vue-app-env.d.ts

```diff
@@ -0,0 +1,64 @@
+declare namespace NodeJS {
+  interface Process{
+    env: ProcessEnv
+  }
+  interface ProcessEnv {
+    /**
+     * By default, there are two modes in Vite:
+     * 
+     * * `development` is used by vite and vite serve
+     * * `production` is used by vite build
+     * 
+     * You can overwrite the default mode used for a command by passing the --mode option flag.
+     * 
+     */
+    readonly NODE_ENV: 'development' | 'production'
+  }
+}
+
+declare var process: NodeJS.Process
+
+declare module '*.gif' {
+  const src: string
+  export default src
+}
+
+declare module '*.jpg' {
+  const src: string
+  export default src
+}
+
+declare module '*.jpeg' {
+  const src: string
+  export default src
+}
+
+declare module '*.png' {
+  const src: string
+  export default src
+}
+
+declare module '*.webp' {
+    const src: string
+    export default src
+}
+
+declare module '*.svg' {
+  const src: string;
+  export default src
+}
+
+declare module '*.module.css' {
+  const classes: { readonly [key: string]: string }
+  export default classes
+}
+
+declare module '*.module.scss' {
+  const classes: { readonly [key: string]: string }
+  export default classes
+}
+
+declare module '*.module.sass' {
+  const classes: { readonly [key: string]: string }
+  export default classes
+}
```

Comment:
Are we missing this as well?

```ts
declare module "*.vue" {
  import Vue from "vue";
  export default Vue;
}
```

Reply:
I was testing this branch on my local and realised all the `*.vue` files import will complain about `cannot find module` on `VSCode` which can be fixed but the snippet I shared. I don't know any good way to do type checking at compile time. Sorry.

Reply:
![image](https://user-images.githubusercontent.com/8111351/83612565-26fd0080-a5b5-11ea-9c68-a10eeed95002.png)
![image](https://user-images.githubusercontent.com/8111351/83612862-907d0f00-a5b5-11ea-9099-78f997f84bb9.png)


You should use the [workspace version of TypeScript](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-the-workspace-version-of-typescript)

Reply:
Oh wow, this resolves the issue. Thanks a lot!